oauth2/src/main/java/com/okta/spring/boot/oauth/env/OktaOAuth2PropertiesMappingEnvironmentPostProcessor.java

```diff
@@ -106,36 +110,61 @@ final class OktaOAuth2PropertiesMappingEnvironmentPostProcessor implements Envir
     @Override
     public void postProcessEnvironment(ConfigurableEnvironment environment, SpringApplication application) {
 
+        OIDCMetadata oidcMetadata;
+        try {
+            String issuer = environment.getProperty(OKTA_OAUTH_ISSUER);
+            if (issuer != null) {
+                // TODO check if it is acceptable to do http requests here
```

Comment:
@arvindkrishnakumar-okta do you think its ok to make http calls from this lib?

Reply:
@deepu105 There's nothing against doing so, but we didn't have to do HTTP calls like this earlier since Spring was managing it for us.

Reply:
We were hardcoding these values before, thats why I did this, so shall we remove the hardcoded values and let spring handle all of it?

Reply:
Yes, if possible.. Ideally we'd want Spring to manage this.

Reply:
As I expecetd not setting these values here doesn't work and a lot of tests fail as well, so I assume this would need a rewrite of a lot of classes.

Reply:
keeping the custom stuff based on discussions